search/lib/cmr.js

```diff
@@ -164,9 +155,12 @@ async function stacIdToCmrCollectionId (providerId, stacId) {
     return collectionId;
   }
   const cmrParams = stacCollectionToCmrParams(providerId, stacId);
-  const collections = await findCollections(cmrParams);
+  let collections = [];
+  if (cmrParams) {
+    collections = await findCollections(cmrParams);
+  }
   if (collections.length === 0) {
-    throw new Error(`Collection ${stacId} not found for provider ${providerId}`);
+    return null;
   } else {
     collectionId = collections[0].id;
     myCache.set(stacId, collectionId, 14400);
```

Comment:
Let's move the retention value to a setting or `const` to avoid magic numbers

Reply:
made this a settings parameter adjustable with an envvar